### PR TITLE
Add devcontainer configuration for GitHub Codespaces.

### PR DESCRIPTION
**Pull request by jedieaston**
This commit adds a devcontainer.json file and some associated configuration files to automatically configure a Canvas development environment in GitHub Codespaces (or locally, via the VSCode remote containers extension). This way, all you have to do is click "New Codespace" (or open the repo in VSCode locally), wait ten minutes, and be ready to contribute!

Test Plan:
- Create a new Codespace using a branch with these files. (Or open it in VSCode locally, and follow the prompts to build a devcontainer.)
- Make sure it builds successfully.
- Run "bundle exec rails db:initial_setup" and "bundle exec rails server" to make sure Canvas comes up successfully.

### CODE REVIEW

File: .devcontainer/postCreate.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+/usr/lib/postgresql/12/bin/initdb ~/postgresql-data/ -E utf8
+/usr/lib/postgresql/12/bin/pg_ctl -D ~/postgresql-data/ -l ~/postgresql-data/server.log start
+/usr/lib/postgresql/12/bin/createdb canvas_development
+# Set correct domain for port forwarding on Codespaces. npx package was developed by a GitHub employee, since you can't get the URL directly in the codespace yet.
+if [ ! -z ${CODESPACES+x} ]
+then
+    sed -i "s/localhost:3000/$(npx -y codespaces-port 3000 | cut -c 9-)/g" config/domain.yml
+fi
+touch /home/vscode/.setupDone
+# bundle exec rails db:initial_setup
```
Comment (octosteve) [inline .devcontainer/postCreate.sh:26]: Why didn't we finish running this here? With Codespaces [Prebuilds](https://docs.github.com/en/codespaces/prebuilding-your-codespaces) you could wind up with an almost instant environment. @jedieaston

Comment (jedieaston) [reply .devcontainer/postCreate.sh:26]: It asked for user input, and at the time I didn't look into how to automate that. I might pick it back up

Comment (sinorrman) [reply .devcontainer/postCreate.sh:26]: Perhaps you found it, but you can predefine those input values with environment variables:
```
[...]
bundle exec rails canvas:compile_assets_dev &&
CANVAS_LMS_STATS_COLLECTION=opt_out \
CANVAS_LMS_ACCOUNT_NAME=School \
CANVAS_LMS_ADMIN_EMAIL=test@school.edu \
CANVAS_LMS_ADMIN_PASSWORD=password12345 \
bundle exec rake db:initial_setup
```
And if you want to run this in Prebuild as @octosteve suggests (and me too), just make sure to run `postCreate.sh` with "**on**CreateCommand" and not "**post**CreateCommand" in the `devcontainer.json`

Comment (octosteve) [reply .devcontainer/postCreate.sh:26]: https://github.com/instructure/canvas-lms/pull/2038/files#diff-c26edb4d2e5f940d49820200be62a92f37f0e2c40c85d237207989d5d6d0d64fR10-R13 For an example.

Comment (jedieaston) [reply .devcontainer/postCreate.sh:26]: I'm going to close this since @octosteve's is farther along. Thanks for the tip though!

**Pull request closed without merging**